src/main/java/io/vertx/ext/jdbc/impl/JDBCClientImpl.java

```diff
@@ -237,7 +240,24 @@ public Future<SQLConnection> getConnection(ContextInternal ctx) {
           Connection conn = holder.dataSource.getConnection();
           Object execMetric = enabled ? holder.metrics.begin(queueMetric) : null;
           // wrap it
-          res.complete(new JDBCConnectionImpl(ctx, helper, conn, holder.metrics, execMetric));
+          String url = conn.getMetaData().getURL();
+          SocketAddress server;
+          Pattern p = Pattern.compile("://(\\p{Alnum}+)(?::([0-9]+))?");
```

Comment:
How about caching in a static final field instead of compiling any a connection is acquired?